src/main/java/com/firebase/geofire/GeoQueryDataEventListener.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Firebase GeoFire Java Library
+ *
+ * Copyright © 2014 Firebase - All Rights Reserved
+ * https://www.firebase.com
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binaryform must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY FIREBASE AS IS AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL FIREBASE BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.firebase.geofire;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+
+/**
+ * GeoQuery notifies listeners with this interface about dataSnapshots that entered, exited, or moved within the query.
+ */
+public interface GeoQueryDataEventListener {
```

Comment:
In order to satisfy #87 we could add another event here `onDataChanged()` that would fire whenever a DataSnapshot within the listener changes.

An `onDataMoved()` event would always be preceded by `onDataChanged()` but it would be possible to see `onDataChanged()` without a subsequent `onDataMoved()`

What do you think?  I imagine that anyone using the DataSnapshot listener would want to know about these events.

Reply:
That does sound like something thats a better fit. I like the approach. I'll implement this here, tomorrow.

Reply:
@vanniktech sweet!

Reply:
Changed it to `An onDataMoved() event would always be preceded by onDataChanged() but it would be possible to see onDataChanged() without an antecedent onDataMoved().` If you know a better word than `antecedent` let me know.